src/graph/immutable_graph.cc

```diff
@@ -449,6 +505,36 @@ ImmutableGraph::EdgeArray ImmutableGraph::EdgeIds(IdArray src_ids, IdArray dst_i
   return ImmutableGraph::EdgeArray{rst_src, rst_dst, rst_eid};
 }
 
+std::pair<dgl_id_t, dgl_id_t> ImmutableGraph::FindEdge(dgl_id_t eid) const {
+  dgl_id_t row = 0, col = 0;
+  auto edge_list = GetEdgeList();
+  CHECK(eid < NumEdges()) << "Invalid edge id " << eid;
+  row = edge_list->src_points[eid];
+  col = edge_list->dst_points[eid];
+  CHECK(row < NumVertices() && col < NumVertices()) << "Invalid edge id " << eid;
+  return std::pair<dgl_id_t, dgl_id_t>(row, col);
+}
+
+ImmutableGraph::EdgeArray ImmutableGraph::FindEdges(IdArray eids) const {
+  CHECK(IsValidIdArray(eids)) << "Invalid edge id array";
+  dgl_id_t* eid_data = static_cast<dgl_id_t*>(eids->data);
+  const auto len = eids->shape[0];
+  std::vector<dgl_id_t> src, dst;
+  for (int64_t k = 0; k < len; k++) {
+    auto edge = ImmutableGraph::FindEdge(eid_data[k]);
+    src.push_back(edge.first);
+    dst.push_back(edge.second);
+  }
+
+  IdArray rst_src = IdArray::Empty({len}, eids->dtype, eids->ctx);
+  IdArray rst_dst = IdArray::Empty({len}, eids->dtype, eids->ctx);
+  dgl_id_t* rst_src_data = static_cast<dgl_id_t*>(rst_src->data);
+  dgl_id_t* rst_dst_data = static_cast<dgl_id_t*>(rst_dst->data);
+  std::copy(src.begin(), src.end(), rst_src_data);
+  std::copy(dst.begin(), dst.end(), rst_dst_data);
```

Comment:
why do you store data in the vector first and copy them? The length of the vectors is known.